### ISSUE TITLE
Open debug window in release?

### ISSUE THREAD

**Issue by djw8605** [unlhcc/HCCGo#61]
What is the best way to open the debug window?  The possible options:

1. Create special debug releases that automatically open the debug window.
2. Add the option to open the debug window from within any release, for example, in a menu item.

**Comment by robogen** [comment]
Tagging @adamfitzgibbon to see what you think.

We could setup a method to have the F keys enabled, so press F12 and debug dialog opens up.

If we use the menu item method, would we want an in-app menu item or use electrons features for having a 'standard' menu?

**Comment by djw8605** [comment]
I would be fine with F - key.  If we went with a menu, then I would recommend using electron's features for a standard menu item.

**Comment by adamfitzgibbon** [comment]
I don't see a problem with having a debug window in normal releases unless we're worried about the possible security repercussions. F-key would be fine too.

**Comment by robogen** [comment]
Sounds like we're pretty unanimous with the F- key functionality. My only hesitation with a menu item is that takes up real estate for the app, and if its for debugging I don't want it to be the only thing in the menu. Perhaps we could (not move) but replicate some of the in-app navigation in the menu bar? Just so there isn't only a 'help' item in the menu.

Could do some extra key-binding then as well, along the lines of Alt-F4 for submit and/or Alt-F8 for logout then exit, things like that.

**Comment by djw8605** [comment]
Ok, F12 sounds good to me.  Know how to do that key binding? 😄

**Comment by robogen** [comment]
Easy method is to use [electron-localshortcut](https://www.npmjs.com/package/electron-localshortcut), I'll get something setup real quick :+1:

**Comment by robogen** [comment]
Created pull request #62 which has keybinding, though I did end up not using electron-localshortcut as in the end it was a wrapper for builtin electron method 'globalShortcut' so I just ended up using that. Also slight modification to keybinding, instead of 'F12' it's 'Alt+F12'.

Also slight Gruntfile.js modification, now instead of 'grunt package' it's 'grunt packageWin' or 'grunt packageOsx' or 'grunt packageNix' depending on which system you're on. This is so we don't all fight over which script is preset in the Gruntfile.js and there's no uniformity there across repos. That modification changes nothing other than the command for packaging.
